client/state/posts/revisions/reducer.js

```diff
@@ -42,13 +38,11 @@ export function requesting( state = {}, action ) {
 export function revisions( state = {}, action ) {
 	if ( action.type === POST_REVISIONS_RECEIVE ) {
 		const { siteId, postId } = action;
-		return {
-			...state,
+		return merge( {}, state, {
```

Comment:
Thanks for these updates, this is looking really good. Just one last thing, I think `merge` is not suited in this exact place because revision are objects that would be merged recursively here and since the reducer is not really aware of the shape of these objects. So, it's possible to end up with a meged revision object (old and new revision of the same Id) and the intent here is to replace the object completely.

Sorry, If I confused you with my previous comment, I like to use `merge` in reducer of scalar values (like above) but not when the reducers objects with unknown shapes.